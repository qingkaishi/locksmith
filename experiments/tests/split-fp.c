#include <pthread.h>

int x;
pthread_mutex_t l;

void bar(void) {
  x = 3;
}

void (*f)(void) = bar;

void *dostuff(void *a) {
  pthread_mutex_lock(&l);
  f();
  pthread_mutex_unlock(&l);
}

int main() {
  pthread_t t1;
  pthread_mutex_init(&l, NULL);
  pthread_create(&t1, NULL, dostuff, NULL);
  pthread_create(&t1, NULL, dostuff, NULL);
}
