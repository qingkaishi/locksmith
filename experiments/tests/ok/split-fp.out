************************* STARTING *************************

shared: &x:split-fp.c:3
&x:split-fp.c:3 is protected by:
  concrete l:split-fp.c:20
  l:split-fp.c:4
*************************** DONE ***************************
